### PR TITLE
new collections API docs

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Rewrites the collections API docs based on feedback recieved
- Reads less like a walkthrough, and more like a guide. Examples are now standalone (feedback: no one was reading the docs from top to finish. People gravitated to the examples).
- Most explanation now moved into the examples via comments (feedback: no one was reading the words)
- Removed some outdated docs that were tripping people up (ex: `$[collections].astro)

## Testing

- N/A

## Docs

- it docs.

### CODE REVIEW

File: docs/collections.md [hunk 1]
```diff
@@ -126,61 +109,57 @@ function formatDate(date) {
...
+  return {
+    // `routes` defines the total collection of routes. 
+    // A route is represented by a `params` data object (instead of a string).
+    // In this example, we format each letter (ex: "a") to params (ex: {letter: "a"}).
+    routes: allLetters.map(letter => {
+      const params = {letter};
+      return params;
+    }),
+    // `permalink` defines the final URL for each route object defined in `routes`.
+    permalink: ({ params }) => `/pokemon/${params.letter}`,
```
Comment (Hanawa02) [inline docs/collections.md:136]: Not sure if it fits, but it could be that the user intends to change the URL /permalink even if not grouping data like this. So this could become a subsection.

File: docs/collections.md [hunk 2]
```diff
@@ -1,107 +1,90 @@
...
-+         ├── $posts.astro     -> /posts/1, /posts/2, …
-+         └── $tag.astro       -> /tag/:tag/1, /tag/:tag/2, …
-```
 
-💁‍ **Tip**: Any `.astro` filename beginning with a `$` is how it’s marked as a collection.
+- `collection` API Refernece: [collection][collection-api].
+- `createCollection()` API Reference: [createCollection][collection-api]
 
-In each `$[collection].astro` file, we’ll need 2 things:
+## Example: Simple Pagination
```
Comment (Hanawa02) [inline docs/collections.md:40]: If the user wants to have something a page per collection item, the only thing that would need to be changed from this simple pagination example would be the `pageSize`?

**Review by Hanawa02 — COMMENTED** [review]
I added some comments with questions that came to me while reading the docs, didn't really work with the collections yet, so I may be off in some comments.

File: docs/collections.md [hunk 3]
```diff
@@ -1,107 +1,90 @@
...
-
-- `/post/:post`: A single blog post page
-- `/posts/:page`: A list page of all blog posts, paginated, and sorted most recent first
-- `/tag/:tag`: All blog posts, filtered by a specific tag
+```js
+// 1. Create a new file in the `src/pages` directory that starts 
+// with the `$` symbol. This is required to enable the Collections API.
+// 
+// Example: `src/pages/$posts.astro` would be the correct file name 
+// for generating a colleciton pages using the `/posts/...` URL scheme.
```
[the remark this answers is not in the document]
Comment (natemoo-re) [reply docs/collections.md:24]: I don't think we support top-level collections, but I can't remember the reason. This was another note from @FredKSchott's Twitch session, so I think we should clarify this.

[the remark this answers is not in the document]
Comment (FredKSchott) [reply docs/collections.md:24]: Oh yea, I thought that it just wasn't supported but that may have been outdated

File: docs/collections.md (hunk 1 above)
Comment (FredKSchott) [reply docs/collections.md:136]: I'm not sure if we really want to encourage this (vs. placing the collection file where you want it) since it breaks with file-based routing a bit

Comment (FredKSchott) [reply docs/collections.md:136]: Maybe I'll just add that as a note to the docs on this line

File: docs/collections.md (hunk 2 above)
Comment (FredKSchott) [reply docs/collections.md:40]: yes! Great call, that honestly should be an example of its own

Comment (FredKSchott) [reply docs/collections.md:40]: added!

File: docs/collections.md [hunk 4]
```diff
@@ -1,107 +1,90 @@
 # 🍱 Collections
 
-## ❓ What are Collections?
+## What are Collections?
 
-[Fetching data is easy in Astro][docs-data]. But what if you wanted to make a paginated blog? What if you wanted an easy way to sort data, or filter data based on part of the URL? Or generate an RSS 2.0 feed? When you need something a little more powerful than simple data fetching, Astro’s Collections API may be what you need.
+Astro Collections help you break up a larger set of data into multiple pages. Examples use-cases include:
```
Comment (drwpow) [inline docs/collections.md:5]:
```suggestion
Astro Collections help you break up a larger set of data into multiple pages. Examples of use-cases include:
```

File: docs/collections.md [hunk 5]
```diff
@@ -1,107 +1,90 @@
...
 
-An Astro Collection is similar to the general concept of Collections in static site generators like Jekyll, Hugo, Eleventy, etc. It’s a general way to load an entire data set. But one big difference between Astro Collections and traditional static site generators is: **Astro lets you seamlessly blend remote API data and local files in a JAMstack-friendly way.** To see how, this guide will walk through a few examples. If you’d like, you can reference the [blog example project][example-blog] to see the finished code in context.
+- Pagination: `/posts/1`, `posts/2`, etc.
+- Grouping Content by Tag: `/tags/red`, `tags/blue`, etc.
+- Grouping Content by Author: `/posts/fred`, `/posts/matthew`, etc.
+- Working with Remote Data
```
Comment (drwpow) [inline docs/collections.md:10]:
```suggestion
- Working with remote data
```

File: docs/collections.md [hunk 6]
```diff
@@ -1,107 +1,90 @@
...
-## ❓ What are Collections?
+## What are Collections?
 
-[Fetching data is easy in Astro][docs-data]. But what if you wanted to make a paginated blog? What if you wanted an easy way to sort data, or filter data based on part of the URL? Or generate an RSS 2.0 feed? When you need something a little more powerful than simple data fetching, Astro’s Collections API may be what you need.
+Astro Collections help you break up a larger set of data into multiple pages. Examples use-cases include:
 
-An Astro Collection is similar to the general concept of Collections in static site generators like Jekyll, Hugo, Eleventy, etc. It’s a general way to load an entire data set. But one big difference between Astro Collections and traditional static site generators is: **Astro lets you seamlessly blend remote API data and local files in a JAMstack-friendly way.** To see how, this guide will walk through a few examples. If you’d like, you can reference the [blog example project][example-blog] to see the finished code in context.
+- Pagination: `/posts/1`, `posts/2`, etc.
+- Grouping Content by Tag: `/tags/red`, `tags/blue`, etc.
+- Grouping Content by Author: `/posts/fred`, `/posts/matthew`, etc.
```
Comment (drwpow) [inline docs/collections.md:9]:
```suggestion
- Grouping Content by author: `/posts/fred`, `/posts/matthew`, etc.
```

File: docs/collections.md [hunk 7]
```diff
@@ -1,107 +1,90 @@
...
 
-## ❓ What are Collections?
+## What are Collections?
 
-[Fetching data is easy in Astro][docs-data]. But what if you wanted to make a paginated blog? What if you wanted an easy way to sort data, or filter data based on part of the URL? Or generate an RSS 2.0 feed? When you need something a little more powerful than simple data fetching, Astro’s Collections API may be what you need.
+Astro Collections help you break up a larger set of data into multiple pages. Examples use-cases include:
 
-An Astro Collection is similar to the general concept of Collections in static site generators like Jekyll, Hugo, Eleventy, etc. It’s a general way to load an entire data set. But one big difference between Astro Collections and traditional static site generators is: **Astro lets you seamlessly blend remote API data and local files in a JAMstack-friendly way.** To see how, this guide will walk through a few examples. If you’d like, you can reference the [blog example project][example-blog] to see the finished code in context.
+- Pagination: `/posts/1`, `posts/2`, etc.
+- Grouping Content by Tag: `/tags/red`, `tags/blue`, etc.
```
Comment (drwpow) [inline docs/collections.md:8]:
```suggestion
- Grouping Content by tag: `/tags/red`, `tags/blue`, etc.
```

File: docs/collections.md [hunk 8]
```diff
@@ -1,107 +1,90 @@
...
-Because `/post/:post` references the static files we have already, that doesn’t need to be a collection. But we will need collections for `/posts/:page` and `/tag/:tag` because those will be dynamically generated. For both collections we’ll create a `/src/pages/$[collection].astro` file. This is our new structure:
 
-```diff
-  └── src/
-      └── pages/
-          ├── post/
-          │   └── (blog content)
-+         ├── $posts.astro     -> /posts/1, /posts/2, …
-+         └── $tag.astro       -> /tag/:tag/1, /tag/:tag/2, …
-```
```
Comment (drwpow) [inline docs/collections.md:37]: I thought the file structure -> URL mapping was helpful, since we’re still using examples. Could this be re-added?

File: docs/collections.md [hunk 9]
```diff
@@ -1,107 +1,90 @@
...
+- `collection` API Refernece: [collection][collection-api].
```
Comment (drwpow) [inline docs/collections.md:37]:
```suggestion
- `collection` API Reference: [collection][collection-api].
```

**Comment by drwpow** [comment]
> Reads less like a walkthrough, and more like a guide

What was the primary change here? I understand updating the outdated parts and expanding this guide, but many sections feel rewritten just for the sake of being rewritten.

**Comment by FredKSchott** [comment]
@drwpow check out: https://documentation.divio.com/
We followed this system of thinking about docs in Snowpack and it worked well, would love to bring this into Astro as well as we start building out the docs site. 

> many sections feel rewritten just for the sake of being rewritten.

Definitely check out the notes and watch the video from the Twitch stream to get an idea of what I'm talking about in the "feedback" comments above: https://gist.github.com/FredKSchott/cb3e34378e5684cb32629b02fdf1506f. A couple of things (all mentioned in OP) were driving the rewrite, but a big one was that sections and examples should be easy to jump around, copy from, etc. Previous examples were diffs = objectively hard to copy paste from, assumed you were reading the entire doc from top to bottom.

File: docs/collections.md (hunk 8 above)
Comment (FredKSchott) [reply docs/collections.md:37]: Yup, tried to move this to this snippet in the API:

```
// Example: `src/pages/$posts.astro` would be the correct file name 
// for generating a collection pages using the `/posts/...` URL scheme.
```

But agreed that's way too wordy. Will rewrite to match your original format

File: docs/collections.md [hunk 10]
```diff
@@ -1,107 +1,90 @@
 # 🍱 Collections
 
-## ❓ What are Collections?
+## What are Collections?
 
-[Fetching data is easy in Astro][docs-data]. But what if you wanted to make a paginated blog? What if you wanted an easy way to sort data, or filter data based on part of the URL? Or generate an RSS 2.0 feed? When you need something a little more powerful than simple data fetching, Astro’s Collections API may be what you need.
+Astro Collections help you break up a larger set of data into multiple pages. Examples of use-cases include:
 
-An Astro Collection is similar to the general concept of Collections in static site generators like Jekyll, Hugo, Eleventy, etc. It’s a general way to load an entire data set. But one big difference between Astro Collections and traditional static site generators is: **Astro lets you seamlessly blend remote API data and local files in a JAMstack-friendly way.** To see how, this guide will walk through a few examples. If you’d like, you can reference the [blog example project][example-blog] to see the finished code in context.
+- Pagination: `/posts/1`, `posts/2`, etc.
```
Comment (drwpow) [inline docs/collections.md:7]:
```suggestion
- Pagination: `/posts/1`, `/posts/2`, etc.
```

File: docs/collections.md [hunk 11]
```diff
@@ -1,107 +1,90 @@
...
+- Grouping content by author: `/posts/fred`, `/posts/matthew`, etc.
```
Comment (drwpow) [inline docs/collections.md:8]:
```suggestion
- Grouping content by author: `/author/fred`, `/author/matthew`, etc.
```

Minor thing, but it should hint that this can‘t coexist with pagination easily (probably not without some gotchas)

File: docs/collections.md [hunk 12]
```diff
@@ -1,107 +1,90 @@
...
+- Grouping content by some tag: `/tags/red`, `tags/blue`, etc.
```
Comment (drwpow) [inline docs/collections.md:9]:
```suggestion
- Grouping content by some tag: `/tags/red`, `/tags/blue`, etc.
```

File: docs/collections.md [hunk 13]
```diff
@@ -126,61 +109,100 @@ function formatDate(date) {
 </html>
 ```
 
-Let’s walk through some of the key parts:
-
-- `export let collection`: this is important because it exposes a prop to the page for Astro to return with all your data loaded. ⚠️ **It must be named `collection`**.
```
Comment (drwpow) [inline docs/collections.md:131]: Doesn’t have to be here exactly, but I think this is an important point to reiterate: this prop must be exported to work

File: docs/collections.md [hunk 14]
```diff
@@ -126,61 +109,100 @@ function formatDate(date) {
...
-- `export async function createCollection()`: this is also required, **and must be named this exactly.** This is an async function that lets you load data from anywhere (even a remote API!). At the end, you must return an object with `{ data: yourData }`. There are other options such as `pageSize` we’ll cover later.
```
Comment (drwpow) [inline docs/collections.md:132]: Should this also be called out that `createCollection` is a reserved word?

File: docs/collections.md [hunk 15]
```diff
@@ -126,61 +109,100 @@ function formatDate(date) {
...
-- `allPosts.filter((post) => post.tag === params.tag)`: we aren’t returning all posts here; we’re only returning posts with a matching tag. _What tag,_ you ask? The `routes` array has `[{ tag: 'javascript' }, { tag: '…`, and all the routes we need to gather. So we first need to query everything, but only return the `.filter()`ed posts at the very end.
+// Define a `createCollection` function. 
+// In this example, we'll customize the URLs that we generate to
+// create a new page to group every pokemon by first letter of their name. 
+export async function createCollection() {
+  const allPokemonResponse = await fetch(`https://pokeapi.co/api/v2/pokemon?limit=150`);
+  const allPokemonResult = await allPokemonResponse.json();
+  const allPokemon = allPokemonResult.result;
+  const allLetters = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'];
```
Comment (drwpow) [inline docs/collections.md:126]: This is a cool example! I dig it

File: docs/collections.md [hunk 16]
```diff
@@ -1,107 +1,90 @@
...
-└── src/
-    └── pages/
-        └── post/
-            └── (blog content)
-```
+To create a new Astro Collection, you must do three things:
 
-The first step in adding some dynamic collections is deciding on a URL schema. For our example website, we’re aiming for the following URLs:
+```js
+// 1. Create a new file in the `src/pages` directory that starts
```
Comment (drwpow) [inline docs/collections.md:21]: I think steps 1, 2, and 3 should live in plain text and not a code block. Step 1 actually doesn’t relate to code at all, and would be easier to read in a paragraph IMO.

Steps #2 and #3 do relate to code, but I think more context is needed to describe where these go in a file. Should these be a diff?

Comment (FredKSchott) [reply docs/collections.md]: SGTM

**Comment by FredKSchott** [comment]
Responded to all comments. Merging this now to replace what's currently there, but keep leaving comments, happy to take more edits in a followup PR.
